### DISCUSSION TITLE
API appeared! But how does it work?

### DISCUSSION THREAD

**Discussion by avchorniy** [haotian-liu/LLaVA#1382] [Q&A]
Hi! I noticed that APIs have appeared, but it works strangely, in particular, there are even taking "examples". If you write "Hello!!!" as an input in a chatbox and upload a picture of a red bus from the example as an image, the response is: "Hello! It looks like you've shared an image of a cartoon-style red bus with a smiling face. It's a simple, friendly representation of a bus, often used in educational materials or as a mascot. If you have any questions about the image or need information related to it, feel free to ask!". However, when I run the code 
```
from gradio_client import Client

client = Client("https://llava.hliu.cc/")
result = client.predict(
		"Hello!!",	# str  in 'parameter_3' Textbox component
		"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# filepath  in 'parameter_11' Image component
		"Crop",	# Literal['Crop', 'Resize', 'Pad', 'Default']  in 'Preprocess for non-square image' Radio component
		api_name="/add_text"
)
print(result)
```
An output is:
`([['<img src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCABEAD0DASIAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6/8QAHwEAAwEBAQEBAQEBAQAAAAAAAAECAwQFBgcICQoL/8QAtREAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9oADAMBAAIRAxEAPwD3+ikZlRSzEBQMkk8CuX1fxpa2m6KwAuZRxv8A4B/j+H51Mpxirs3oYarXly01c6We4htYWmnlSONerOcCuP1fxwBui0tM9vPkH8h/j+Vcnf6neanN5t3M0h7L0VfoKqVx1MS3pHQ+iwmTU6fvVvefbp/wTpNN8Z6haPtusXURPO7hh9D/AI12mma/p+rAC3mAl7xPw35d/wAK8npQSpBBII5BFTCvKO+pvicpoVtYrlfl/ke1UV5vpfjG/sdsdz/pUI/vn5x9G7/jXcaRrVlrds81nIW8ttkinqjYzg/gRXXCrGex87isvrYbWSuu6PKvGnjG7h8V32l3TMbKB1CLHxjKg8jvyap213Bdx74JVcd8dR9R2rU1KwtL7xn4kS6t45R5sQ+Ycj5B0PUVzuoeDp7aT7RpE7Ajny2bDfgf8axlgcTOHt4e8nfTqrNo9GjxVkuHxX9mYh+xnFQ95/BJyjGWr6PXrp5mpRXOw67c2U32fVLd1YfxbcN+I7/hWs2rWC2v2hrqMReuefpjrmuFSV7dT6mVKSiprWL2a1T9GXKpX+q2emx7rmUBscIOWP0Fc3f+Kri6k+z6XEy7uA+3Ln6DtWcmjzz6bJqdzOck8Kclm+bByTWkKc6l+RXtr9xz18RQwvK8RK12kl1bexau/EOo6vOLWwRolfgBD8zfU9q9a+DdrLZaHqlvMQZFvAWwc9Y0P9a4L7JBaahYxwRLGoWToPYV6T8MGVrbWipBH21Rkf8AXJK9P6ksOuaTvK9vLa58fWz95jenThy0+Tm878/Lq9tun4nGeIdfGi/ELW1khMkMske7afmXCDkevWtiw1Wy1OPfazq5AyV6Mv1FcX8RP+R91b/rov8A6Atc3FNJBIskMjRuvRlOCK5cNm9XCzdNq8bv13O7OPD3A55ShiYT9nXcI67qVopar7ldNeaZ65eWNrfw+VdQJKvbcOR9D2rznXPD0Fv4ltdPilcQzIZMnkqOcj/x2tLS/G1xDtj1CPz0/wCeicOPw6H9Kra5rVhP4p0++jmLW6QFWbacg/Nxj8RXo4nE4LFxjNWvdXvo7dfkfLZDknEvD2KrYWfM6Tpza5feg5Je60ujb2uk35kuk2MFhql7FCp2qkWCxyed2f5VSur+2h8OGEyqZmY7UByf9Zn8KzL3xBM97cy2RMSTBVJIG7Az+XU1i1zVsfCEXTor+ZeWr6fI+nwPDeIxFVYrHTd/3crbu8I2d+2r6fgamqa5cajMjKPIRAQoRjnnrk17F8Dv+RY1H/r8/wDZFrxSxt9SnW9k0q0tbiSys5b24N0AUSFB82AerHPH0/EfQXwu8P3Wg+Hbg3Kxqt5P9ogVH3ERlRtyfXArnp+1qVFVm73OrMJ4PD4WWBw8eXkt+d/V+d+5zfxX8K3l9q1hf6PaRIzErfSyZCuDt2tkdxgjpzkelefaj4fvtOy5j82Ef8tI+QPqO1fTpAZSrAEHgg96wdR8KWd3l7b/AEaU9lGVP4dvwpYjCKo+ZbnPl+e18MlCTvFbeXo+npqvI+a6rXYJMYHU5r1nxD4D2MzywGBj0nhGUb6j/wDUa8813wtqdtp95MgR44YJH3q3J47DrmvPVCcZpM+ujnWFq4adTm962zt06Luc9HJBOpNtcJPhtrbAeD+I5z2Psa6fSPBt5e7ZbzNrAecEfO34dvx/Kus8D6Tb+KdTvda0PRRY6U0dvb27SxBOYkIdxjPOWxkZJxz3r1rTPDVnYYkkH2icfxOOB9BXb9VvNpbHzMuIKnsUre8eSR/CjV9QvYn0i/Ol6XcWxtNQMmS88O7cQoxzn8B9ele0aXpkGk2a21uXKL/ebP5DoPwq7RXbCCikux89VqyqScpdQoooqzIQgMpVgCDwQe9ZTeG9Le7Fwbce8YPyE+uP8iiipkk9y4yavZmpHGkUaxxoqIowqqMAD2FOooqiAooooA//2Q==" alt="user upload image" />Hello!!', None]], '', None)`
That is, in fact, the answer does not include a response from the chatbot. 
Can anyone help me understand these APIs? I want to upload an image, ask a question about it and get a response from the chatbot.

**Comment by zty0510** [comment]
You should then call '/http_bot' api
